runtime/libia2/init.c

```diff
@@ -15,6 +16,50 @@
 extern __thread void *ia2_stackptr_0[PAGE_SIZE / sizeof(void *)]
     __attribute__((aligned(4096)));
 
+/// The TLS pthread key for storing thread stacks.
+/// The value of the data set with `pthread_setspecific` is not used;
+/// it just matters if it was set (to non-`NULL`) or not.
+static pthread_key_t thread_stacks_key IA2_SHARED_DATA;
+
+/// The compartment stacks for each thread.
+static __thread void *stacks[IA2_MAX_COMPARTMENTS] = {0};
+
+/// The destructor for `thread_stacks_key`.
+/// It deallocates all of the compartment stacks for the current thread being destructed.
+static void thread_stacks_destructor(void *_unused) {
+#if IA2_VERBOSE
+  char thread_name[16] = {0};
```

Comment:
Please add a comment explaining the 16. It wasn't immediately obvious to me that that wasn't an arbitrary value.

Reply:
Fixed in c66f38cda.  Also fixed it in `memory_maps.c` where I did the same thing.